File: 20242Q/poo/laboratorio/tp03/ej9/src/Ellipse.java
public class Ellipse extends Figure{

    Point centerPoint;
    // los defino como radio, pero puede que esté mal por cómo se calcula area y perim de elipse
    double mayorRadius;
    double minorRadius;

    public Ellipse(Point cp, double mayr, double minr) {
        centerPoint = cp;
        mayorRadius = mayr;
        minorRadius = minr;
    }

    @Override
    public double area() {
        return Math.PI / 4 * mayorRadius * minorRadius;
    }

    @Override
    public double perimeter() {
        return Math.PI / 2 * (mayorRadius + minorRadius);
    }

    @Override
    public String toString() {
        return "Elipse [Centro: %s, DMayor: %.2f, DMenor: %.2f]".formatted(centerPoint, mayorRadius, minorRadius);
    }
}

File: 20242Q/poo/laboratorio/tp08/ej4/ellipse.rb
require_relative 'figure'

class Ellipse < Figure
  def initialize(center, mayor_axis, minor_axis)
    @center = center
    @mayor_axis = mayor_axis
    @minor_axis = minor_axis
  end

  def area
    Math::PI / 4 * @mayor_axis * @minor_axis
  end

  def perimeter
    Math::PI / 2 * (@mayor_axis * @minor_axis)
  end

  def to_s
    "Elipse [Centro: #{@center}, DMayor: #{@mayor_axis}, DMenor: #{@minor_axis}]"
  end
end
